server/plugins/ansible/impl/src/main/java/com/walmartlabs/concord/server/plugins/ansible/ProcessAnsibleResource.java

```diff
@@ -327,9 +329,18 @@ public List<AnsibleHostEntry> list(UUID instanceId, OffsetDateTime instanceCreat
                 if (playbookId != null) {
                     q.and(a.PLAYBOOK_ID.eq(playbookId));
                 }
-
-                return q.orderBy(a.HOST)
-                        .limit(limit)
+                
+                if (sortField != null) {
+                    if (sortBy != null && sortBy.equals(SortBy.DESC)) {
+                        q.orderBy(Objects.requireNonNull(a.field(sortField.name())).desc());
```

Comment:
Maybe add a message to `requireNonNull`, so if it fails it will be more apparent on which field it failed.
E.g.
```java
Field orderBy = requireNonNull(a.field(sortField.name()), "Invalid sort field: " + sortField.name()).desc();
q.orderBy(orderBy)
``` 

I think it is even possible to do such validation outside of the transaction.